pfinance/functions.py

```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
     return (purchase_price - salvage_value) / useful_life
 
 
+def double_declining_balance_depreciation(
+    purchase_price: float,
+    salvage_value: float,
+    useful_life: int,
+    factor: float = 2.0,
+) -> dict[list[float], list[float]]:
+    '''
+    Calculate the depreciation of an asset using double declining balance.
+
+        Parameters:
+            purchase_price (float): The total amount paid for the asset
+            salvage_value (float): The value of the asset after its useful life
+            uselife_life (int): The expected lifespan of an asset, must be greater than 0
+            factor (int): The rate at which the balance declines, default 2
+
+        Returns:
+            double_declining_balance_result (dict):
+                asset_value (list[float]): Value of the asset at beginning of the period
+                periodic_depreciation (list[float]): Deprecriation of the asset at end of the period
+    '''
+    asset_value = [purchase_price * 1.0]
```

Comment:
What's the purpose of the `* 1.0`?

Reply:
I was thinking of keeping all the values inside asset_value as a float which is why I multiplied by 1.0 (but now that I think about it, i could just cast to float instead).

Reply:
Either is fine with me, even if you just decide it doesn't need to be a float that would work too (looks better as a float for display purposes but any division, etc would treat an int as a float anyways)